packages/material-ui/src/NoSsr/NoSsr.js

```diff
@@ -34,28 +32,27 @@ class NoSsr extends React.Component {
         requestAnimationFrame(() => {
           // The UI is up-to-date at this point.
           // We can continue rendering the children.
-          if (this.mounted) {
-            this.setState({ mounted: true });
+          if (mountedRef.current) {
+            setMountedState(true);
           }
         });
       });
     } else {
-      this.setState({ mounted: true });
+      setMountedState(true);
     }
-  }
-
-  componentWillUnmount() {
-    this.mounted = false;
-  }
 
-  render() {
-    const { children, fallback } = this.props;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, [defer]);
 
-    return this.state.mounted ? children : fallback;
-  }
+  return <React.Fragment>{mountedState ? children : fallback}</React.Fragment>;
```

Comment:
Looks like react-docgen can't tell it's a React component 😕 

returning a fragment fixes this

Reply:
Can we add a comment in the code to explain why?